### PR TITLE
Added support for Build Multiple Versions of React Native app

### LINKED ISSUE #764: Can't debug if build configuration is different than Debug
I followed this article https://medium.com/@ywongcode/building-multiple-versions-of-a-react-native-app-4361252ddde5 to have different build variants pointing to different environments (e.g. dev, staging, prod).

I have these 4 build configurations:
- Debug
- Release
- Dev.Debug
- Dev.Release

and these 2 schemes:
- appname (using Debug and Release)
- appname-dev (using Dev.Debug and Dev.Release)

Dev.Debug and Dev.Release have a different bundle id (com.appname.dev) than Debug and Release (com.appname).

I manage to make everything install and run fine using this VS code extension using `runArguments` in `launch.json` like so:

```
{
  "name": "Debug iOS - iPhone X - Dev",
  "program": "${workspaceRoot}/.vscode/launchReactNative.js",
  "type": "reactnative",
  "request": "launch",
  "platform": "ios",
  "sourceMaps": true,
  "outDir": "${workspaceRoot}/.vscode/.react",
  "runArguments": [
    "--simulator=iPhone X",
    "--scheme=appname-dev",
    "--configuration=Dev.Debug"
  ]
},
{
  "name": "Debug iOS - iPhone X - Prod",
  "program": "${workspaceRoot}/.vscode/launchReactNative.js",
  "type": "reactnative",
  "request": "launch",
  "platform": "ios",
  "sourceMaps": true,
  "outDir": "${workspaceRoot}/.vscode/.react",
  "runArguments": [
    "--simulator=iPhone X",
    "--scheme=appname",
    "--configuration=Debug"
  ]
}
```

The only issue is that the debugger doesn't seem to work properly and I see this error in the DEBUG CONSOLE tab:

```
Could not debug. Error while executing command '/usr/libexec/PlistBuddy -c 'Print:CFBundleIdentifier' '/Users/simonreggiani/dev/appname/ios/build/Build/Products/Debug-iphonesimulator/appname.app/Info.plist'': Command failed: /usr/libexec/PlistBuddy -c 'Print:CFBundleIdentifier' '/Users/simonreggiani/dev/appname/ios/build/Build/Products/Debug-iphonesimulator/appname.app/Info.plist'
Print: Entry, ":CFBundleIdentifier", Does Not Exist
```

The issue is that `/Users/simonreggiani/dev/appname/ios/build/Build/Products/Debug-iphonesimulator/appname.app/Info.plist` doesn't exist.
But `/Users/simonreggiani/dev/appname/ios/build/Build/Products/Dev.Debug-iphonesimulator/appname.app/Info.plist` does exist.

The difference is the build configuration `Dev.Debug` instead of `Debug`.

Would it be possible to have a param of the `reactnative` launch type that would specify the build configuration?

If that makes sense, I'd be happy to send a PR with that change.

## Actual Behavior
1. Have a project with different build configurations in Xcode
2. Launch debug
3. Debug doesn't work

## Expected Behavior
1. Have a project with different build configurations in Xcode
2. Launch debug
3. Debug works

## Software versions
- React-Native VS Code extension version: 
- VSCode version: 1.25.1
- OS platform and version: 0.6.13
- NodeJS version: v8.11.3
- React Native version: 0.56.0

### PR DESCRIPTION
**Pull request by iTOYS**
Resolve #764

### CODE REVIEW

File: src/extension/ios/plistBuddy.ts [hunk 1]
```diff
@@ -3,33 +3,41 @@
...
     }
 
-    public getBundleId(projectRoot: string, simulator: boolean = true): Q.Promise<string> {
-        return this.xcodeproj.findXcodeprojFile(projectRoot).then((projectFile: IXcodeProjFile) => {
-            const infoPlistPath = path.join(projectRoot, "build", "Build", "Products",
-                simulator ? "Debug-iphonesimulator" : "Debug-iphoneos",
-                `${projectFile.projectName}.app`, "Info.plist");
+    public getBundleId(projectRoot: string, simulator: boolean = true, configuration: string = "Debug", productName?: string): Q.Promise<string> {
+        const productsFolder = path.join(projectRoot, "build", "Build", "Products");
+        const configutationFolder = path.join(productsFolder, `${configuration}${simulator ? "-iphonesimulator" : "-iphoneos"}`);
```
Comment (ruslan-bikkinin) [inline src/extension/ios/plistBuddy.ts]: Seems to be that there is a typo: `configutationFolder`.

File: src/extension/ios/iOSPlatform.ts [hunk 2]
```diff
@@ -53,15 +53,19 @@ export class IOSPlatform extends GeneralMobilePlatform {
     constructor(protected runOptions: IIOSRunOptions, platformDeps: MobilePlatformDeps = {}) {
         super(runOptions, platformDeps);
 
+        const configurationIdx = this.runArguments.indexOf("--configuration");
+        const configuration = configurationIdx > -1 ? this.runArguments[configurationIdx + 1] : "";
```
Comment (sregg) [inline src/extension/ios/iOSPlatform.ts]: Are you sure `configurationIdx + 1` is what you wanted here?

Comment (sregg) [reply src/extension/ios/iOSPlatform.ts]: I tried it and got this:
```
Could not debug. Could not found executable in /Users/simonreggiani/app/ios/build/Build/Products/-iphonesimulator
```

with this `launch.json`:

```
{
  "name": "Debug iOS - iPhone X - Dev",
  "program": "${workspaceRoot}/.vscode/launchReactNative.js",
  "type": "reactnative",
  "request": "launch",
  "platform": "ios",
  "sourceMaps": true,
  "outDir": "${workspaceRoot}/.vscode/.react",
  "runArguments": [
    "--simulator=iPhone X",
    "--scheme=app-dev",
    "--configuration=Dev.Debug"
  ]
}
```

**Comment by freakinruben** [comment]
@iTOYS I can't figure out how I can specify the `productName` in the `launch.json`, should this be passed on in the `runArguments`?

**Comment by freakinruben** [comment]
I see that it's reverted for some reason in 0.16.6 and added again in master (unreleased). That's probably why I can't get it to work.. @ruslan-bikkinin any chance this can be released again?

**Comment by ruslan-bikkinin** [comment]
Hi @freakinruben, we are going to release new version at the beginning of the next week. I'll keep you posted.

**Pull request merged**